lib/adagrams.rb

```diff
@@ -0,0 +1,112 @@
+  require 'csv'
+
+  def draw_letters
+    alphabet = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1
+    }
+
+    hand = []
+    tiles_drawn = 0
+    until tiles_drawn == 10
+      letter = alphabet.keys.sample
+      if alphabet[letter] > 0
```

Comment:
Using a hash with sample gives an equal 1/26 chance to each letter, instead we need the chances to be weighted.